### PR TITLE
Vendor to gardener v1.16.0 and update controller runtime to v0.7.1

### PR DESCRIPTION
**Pull request by DockToFuture**
**How to categorize this PR?**

/area networking
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Vendor to gardener v1.16.0 and update controller runtime to v0.7.1

**Release note**:

```other operator
Vendor to gardener v1.16.0 and update controller runtime to v0.7.1
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ############# builder
-FROM golang:1.15.3 AS builder
+FROM golang:1.15.7 AS builder
```
Comment (vpnachev) [inline Dockerfile:2]:
```suggestion
FROM eu.gcr.io/gardener-project/3rd/golang/golang:1.15.7 AS builder
```

Please, update also the alpine image. you can use `eu.gcr.io/gardener-project/3rd/alpine:3.12.3`.

**Review by vpnachev — COMMENTED** [review]
One proposal regarding docker images.
Rest is looking good to me.

File: Dockerfile (hunk 1 above)
Comment (DockToFuture) [reply Dockerfile:2]: you probably mean `FROM eu.gcr.io/gardener-project/3rd/golang:1.15.7 AS builder` sure will update it.

**Review by vpnachev — APPROVED** [review]
/lgtm
/needs/second-opinion

**Review by rfranzke — APPROVED** [review]

**Pull request merged**